xmppserver/src/main/java/org/jivesoftware/openfire/group/DefaultGroupPropertyMap.java

```diff
@@ -242,8 +261,14 @@ public boolean remove(Object o) {
          */
         @Override
         public void clear() {
+            final Map<String,String> originalMap = new HashMap<>(); // copy to be used by event handling.
+            for (Entry<K,V> entry : delegate) {
+                if (entry.getKey() instanceof String && entry.getValue() instanceof String) {
+                    originalMap.put((String) entry.getKey(), (String) entry.getValue());
+                }
+            }
             delegate.clear();
-            deleteAllProperties();
+            deleteAllProperties(originalMap);
```

Comment:
This method doesn't do what the Javadoc says ("Removes all the elements in the set") - possibly "Removes all the elements in the set whose values are Strings" ?

Reply:
Actually, they _are_ all deleted. The map that's passed in is only used for event dispatching. I'm unsure if there is any sense in having this be a generic map - if we're not using Strings, this implementation seems to choke.